### ISSUE TITLE
font attributes leak past \textbf{}

### ISSUE THREAD

**Issue by xworld21** [brucemiller/LaTeXML#1625]
Something unexpected that I am not able to debug myself:
```latex
\documentclass{article}
\usepackage{graphicx}
\begin{document}
\textbf{Bold text.}\includegraphics{test.svg}
\end{document}
```
generates
```xml
<p><text font="bold">Hello.<graphics graphic="test.svg" xml:id="p1.g1"/></text></p>
```
even though I'd expect `<graphics>` to be outside.

I am not aware of real-life documents where this is a problem, but it is odd. I can say it interferes with certain experiments I am doing with `<ltx:rawliteral>` - I am not able to place the literal where I need because sometimes it ends up within a `<text>`.

**Comment by dginev** [comment]
Right, so if you add some text after it will close correctly and keep it in a new font:
```tex
\textbf{Bold text.}\includegraphics{test.svg} odd?
```
```xml
<p><text font="bold">Bold text.<graphics graphic="test.svg" xml:id="p1.g1"/></text> odd?</p>
```

which is clearly different than the \bfseries continuation:
```tex
\bfseries Bold text. \includegraphics{test.svg} bold here too.
```
```xml
    <p><text font="bold">Bold text. <graphics graphic="test.svg" xml:id="p1.g1"/> bold here too.</text></p>
```

The key discussion is what exactly is the relationship between the graphics element, and other non-textual pieces that can be found inline, and the carrying `ltx:text` element. There's already some confusion (also for me) based on the names, as finding graphics inside something called `text` is clearly a shocker. Whether that ought to have been `ltx:inline` or `ltx:span`, I'm unsure.

The schema mastermind here is certainly @brucemiller , so he may want to elaborate on the relationships here.

**Comment by brucemiller** [comment]
It is interesting that the 3 of us look at the case and perceive 3 distinctly different bugs :>
It's kinda hard to know how to address it, since even the simple demonstration bug isn't quite the case that @xworld21 is really interested in.

It may well be the case that `ltx:text` is misleading or misnamed. It indeed can be thought of as like HTML's `span`, so it can contain any inline or phrase markup, including `ltx:graphic` or for that matter `ltx:rawliteral`.

On the other hand, while I often see `\textbf{...}` and am expecting a straightforward `<ltx:text>...</ltx:text>` to contain the argument, in fact, all the `\textXX` commands are trying to cooperate with the `\bf` (etc) commands to switch fonts & minimally contain the chunks in different fonts. That's old & magic code and hard to "improve" without breaking lots of things.  It would seem that having `ltx:graphics` use a default font would cause that magic machinery to close the `\textbf`, but turns out to be a bit harder to get to work --- and then I realized that fixating on `ltx:graphics` doesn't necessarily solve the problem @xworld21 was having anyway.  While we *might* think that a graphic shouldn't inherit the font, I can't think of what assumption on the font of `ltx:rawliteral` could possibly make sense.

In short, I'm not sure what the fix or workaround should be, other than adding some innocuous space or something as @dginev suggested.

**Comment by xworld21** [comment]
Re `ltx:rawliteral`, I have learned that my tentative approach was quite misguided, so I ended up adding my own elements to the schema, which was the 'correct' way of doing what I needed anyway.

From my point of view, the true issue is that on one hand `\textbf` opens and closes a TeX group, while on the other LaTeXML keeps `ltx:text` open for a little longer. That doesn't seem to have real consequences, though, so... you can close this issue, until the grouping mismatch resurfaces again in an actual bug?

**Comment by brucemiller** [comment]
OK, cool.  There very well could be a bug, I just don't know which one to fix :>

And if you have generally useful schema extensions, we could discuss!

Thanks for the report... or rather, the interest in making a report :>
